analyzers/src/SonarAnalyzer.CSharp/Rules/StaticFieldInGenericClass.cs

```diff
@@ -40,41 +41,41 @@ public sealed class StaticFieldInGenericClass : SonarDiagnosticAnalyzer
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
         protected override void Initialize(SonarAnalysisContext context) =>
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
                 {
                     var typeDeclaration = (TypeDeclarationSyntax)c.Node;
-                    if (c.ContainingSymbol.Kind != SymbolKind.NamedType
+                    if (c.IsRedundantPositionalRecordContext()
                         || typeDeclaration.TypeParameterList == null
                         || typeDeclaration.TypeParameterList.Parameters.Count < 1)
```

Comment:
```suggestion
                        || c.Node is not TypeDeclarationSyntax { TypeParameterList: { Parameters: { Count: > 0 } } } typeDeclaration)
```

Reply:
@csaba-sagi-sonarsource That's one of the places where C#10 would remove some clutter:
```cs
// C# 9
c.Node is not TypeDeclarationSyntax { TypeParameterList: { Parameters: { Count: > 0 } } } typeDeclaration)
// C# 10
c.Node is not TypeDeclarationSyntax { TypeParameterList.Parameters.Count: > 0  } typeDeclaration)
```